src/JWTGuard.php

```diff
@@ -368,6 +377,21 @@ public function getUser()
         return $this->user;
     }
 
+    /**
+     * Set the current user.
+     *
+     * @param \Illuminate\Contracts\Auth\Authenticatable $user
+     * @return $this
+     */
+    public function setUser(Authenticatable $user)
+    {
+        $this->user = $user;
+
+        $this->fireAuthenticatedEvent($user);
+
+        return $this;
+    }
+
```

Comment:
So it isn't better to move straight to trait and do everything in this PR?

Reply:
sorry, "parser error" 😬 Can you re-phrase maybe?

Reply:
Laravel also overwrites setUser for the same reason: https://github.com/laravel/framework/blob/8.x/src/Illuminate/Auth/SessionGuard.php#L923

Thats why I did it the same way...

The goal was to raise the same events as SessionGuard where it is possible and for setUser a Authenticated event should be raised.

Reply:
@mfn on that case it's seems we're facing a limitation of the Laravel framework for setUser since it's overridden. 

@okaufmann does that would harm the library? As I far I did see any problem at all.

Reply:
Hmm you're right, this is a risk when Laravel changes the behaviour of setUser in its Auth Guards. But this is not a breaking change at all, as it does only raise an event now. When no one listens for this event, nothing happen. Maybe you should note it in the Release notes.

So I think we can risk it. What do you think?

Reply:
well yes, this is what I was insinuating with the alias; to be on the safe side we could do this (untested):
```diff
diff --git a/src/JWTGuard.php b/src/JWTGuard.php
index ba02510..b89a0ab 100644
--- a/src/JWTGuard.php
+++ b/src/JWTGuard.php
@@ -31,7 +31,10 @@ use PHPOpenSourceSaver\JWTAuth\Exceptions\UserNotDefinedException;
 
 class JWTGuard implements Guard
 {
-    use GuardHelpers, Macroable {
+    use GuardHelpers {
+        setUser as guardHelperSetUser;
+    }
+    use Macroable {
         __call as macroCall;
     }
 
@@ -385,11 +388,11 @@ class JWTGuard implements Guard
      */
     public function setUser(Authenticatable $user)
     {
-        $this->user = $user;
+        $result = $this->guardHelperSetUser($user);
 
         $this->fireAuthenticatedEvent($user);
 
-        return $this;
+        return $result;
     }
 
     /**
```
WDYT?

Reply:
Great idea! Seems to work as well.